### PR TITLE
[WIP] Floyd's wild sound adventure (Adds sound_loop datums to a bunch of machinery)

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus:
soundadd: Adds a few sound_loop datums to machinery.
/:cl:

This PR adds sound_loop datums to machinery, meaning that machinery now actually makes sound while active. Currently the 2 machine types affected are Tcomms and computers, But I plan on adding more. The PR is already up because I want some feedback on the sounds that I have so far; and to find out if people think the sounds are too disruptive.

Also; the sounds are currently a bit long range, but I plan on making it so you can only hear it when you are fairly close to the machine

new example vid:
https://streamable.com/4be5v

old example vid: (disclaimer: sounds will not be this long range in the final product)
https://streamable.com/12pi1

pc startup sound:
https://vocaroo.com/i/s0OEuGPgE0Rp

tcomms loop sounds:
https://vocaroo.com/i/s0uFSHxGk7NL

### CODE REVIEW

**Comment by Cuboos** [comment]
You mean to tell me we have looping sounds now?

**Comment by Qustinnus** [comment]
yes they're datums, they're a recent addition from ninjanomnom. you can set a start, a list of loop and an ending sound.

**Comment by Cuboos** [comment]
Well fuck, maybe i'll get back into making sounds for /tg/ station again.

**Comment by Qustinnus** [comment]
lets share sound libraries and make this place a sound heaven :^)

**Comment by Cuboos** [comment]
I don't think i have a file sharing system big enough to share my library...

**Comment by Qustinnus** [comment]
well if you run into any good machinery sounds hit me up, I'm going to try and add sounds to as much machinery as possible. 

that reminds me actually, this could be fucking sick on borgs.

**Comment by Cuboos** [comment]
Well actually... i was working on machinery sounds a little while ago, but quit because there wasn't a good way to make them loop. I already completed sounds for the Autolathe and protolathe.

**Comment by Qustinnus** [comment]
Oh sick! that works out then. the loop sounds need to be the same length btw (in a single datum that is).

**Comment by Cuboos** [comment]
Same length?

**Comment by Qustinnus** [comment]
yeah you need to define the length of the audio for the datum to work (for example if you have 3 looping sounds for the autolathe, and they're all 3 seconds, you'd set mid_length to 3)

**Comment by Cuboos** [comment]
Is it possible to have infinite loops?

**Comment by Qustinnus** [comment]
yes. untill the object the sound loop is attached to gets deleted. you can also turn it off and on whenever you want (for example if the machine is off you can make the sound turn off).

**Comment by Cuboos** [comment]
Welp, time to dust off the ol' git console and see what i can do with this.

**Comment by Qustinnus** [comment]
if you need any help feel free to ping me or @ninjanomnom. we can both give you a hand if you need one. Currently the only thing that's missing which I think Ninja said he would work on was the ability to set a custom range for the sound (because the distance that the sound travels is quite far at the moment)

**Comment by Cuboos** [comment]
that was fast...

**Comment by Qustinnus** [comment]
it wasn't that big a task :P playsound already supports range modifiers, i'm testing right now if it works

**Comment by Ecolli** [comment]
It's subtle and totally immersive. Good job Floyd. 

This was an audio loop I was hoping to make on the ship while it's under stress or something. Maybe you can make something out of it. https://www.youtube.com/watch?v=hVasq6L6gvc&t=15s

**Comment by Qustinnus** [comment]
Could probably use it for something, cheers.

**Comment by Cuboos** [comment]
What about the audio loop i made for the ship?

**Comment by ninjanomnom** [comment]
length is in deci-seconds actually, I just have overlap in sounds played that I added to account for lag and so there's no gaps.

**Comment by ninjanomnom** [comment]
The main reason I made the datum was to have sounds which played forever but not necessarily in the same order so as to prevent that annoying feeling when you've found the break in a repeating sound.

**Comment by AnonymousNow** [comment]
That computer sound is kind of grating.

**Comment by Bawhoppen** [comment]
@Cuboos I thought you died, where have you been?

**Comment by Cuboos** [comment]
@Bawhoppen I've been working on some of my own stuff for a while.

**Comment by XDTM** [comment]
The computer sounds are pretty loud, might get annoying after a while

That said, 10/10 i love more ambience

**Comment by Qustinnus** [comment]
I'm already working on making them less loud and shorter range.

**Comment by Qustinnus** [comment]
https://streamable.com/4be5v

**Comment by Cuboos** [comment]
Mind if i try working on the sound a little bit? I'm pretty good at making sounds work in TG without being annoying... Minus the frog voices but that wasn't my idea.

**Comment by Qustinnus** [comment]
feel free to edit whatever you like :P I've only done some for tcomms computers and borgs so far

File: code/game/machinery/computer/_computer.dm [hunk 1]
```diff
@@ -63,9 +67,13 @@
 /obj/machinery/computer/power_change()
 	..()
 	if(stat & NOPOWER)
+		if(soundloop)
+			soundloop.stop()
 		set_light(0)
 	else
 		set_light(brightness_on)
+		if(soundloop)
+			soundloop.start()
```
Comment (ninjanomnom) [inline code/game/machinery/computer/_computer.dm]: Better to know if for some reason it gets deleted rather than hiding the error.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/_computer.dm (hunk 1 above)
Comment (Qustinnus) [reply code/game/machinery/computer/_computer.dm]: i like to hide my insecurities :^)


but sure I'll go check for any soundloop killing secrets

**Comment by Qustinnus** [comment]
please drop some ideas on what other machines should make noise

**Comment by GunHog** [comment]
Gravitational generator, the Exosuit Fabricator (when building), the cloner (when cloning), the cryogenics pods (when active), and the DNA modifiers (something quiet though) come to mind.

**Comment by Qustinnus** [comment]
sweet, those sound good. I'll work on those next.

**Comment by Cuboos** [comment]
I could do something like that.

**Comment by AnonymousNow** [comment]
[I need this background sound effect somewhere in the game (not the speech, obviously).](https://www.youtube.com/watch?v=cDAjJaM8bxg)

**Comment by MoreRobustThanYou** [comment]
@Ecolli no offense but that sounds like someone who ate too much taco bell

**Comment by Vivalas** [comment]
Atmos equipment / turbine / engineering heavy machinery

**Comment by optimumtact** [comment]
be less WIP

**Comment by Dorsisdwarf** [comment]
This sounds really awful, the whole station is perpetually loud, the bridge is deafening.

**Comment by Dorsisdwarf** [comment]
no matter where you go its like someone in the next flat over is running a hoover or three non-stop

**Comment by Qustinnus** [comment]
>>merging WIP PRs

**Comment by Qustinnus** [comment]
what the hell is this sabotage

**Pull request merged**